File: python/bubble_sort.py
# -*- coding: utf-8 -*-
''' module bubble_sort '''


def bubble_sort(list):
    ''' sort list '''
    for i in range(0, len(list)):
        for j in range(0, len(list) - i - 1):
            if list[j] > list[j + 1]:
                tmp = list[j]
                list[j] = list[j + 1]
                list[j + 1] = tmp
    return list

File: javascript/typescript/bubble_sort.ts

export function bubbleSort(list: number[]) {
	for (let i = 0; i < list.length - 1; i++) {
		for (let j = 0; j < list.length - 1 - i; j++) {
			if (list[j] > list[j + 1]) {
				const tmp = list[j];
				list[j] = list[j + 1];
				list[j + 1] = tmp;
			}
		}
	}
}